lms/djangoapps/course_home_api/outline/v1/tests/test_views.py

```diff
@@ -94,6 +94,27 @@ def test_get_authenticated_user_not_enrolled(self):
         assert all((block.get('title') != '') for block in date_blocks)
         assert all(block.get('date') for block in date_blocks)
 
+    @override_experiment_waffle_flag(COURSE_HOME_MICROFRONTEND, active=True)
+    @override_waffle_flag(COURSE_HOME_MICROFRONTEND_OUTLINE_TAB, active=True)
+    def test_get_authenticated_user_unenrolled(self):
```

Comment:
Okay, this is a lot of duplicated code between the not_enrolled and _unenrolled tests. Maybe use ddt to indicate whether you should unenroll or not enroll and then you don't need to copy all the checks below.

So something like 
```
@ddt.data('unenrolled', 'not_enrolled')
def test_get_authenticated_user_no_active_enrollment(self, enrollment_status):
    if enrollment_status == 'unenrolled':
        # Create an enrollment, then unenroll to set is_active to False
        CourseEnrollment.enroll(self.user, self.course.id)
        CourseEnrollment.unenroll(self.user, self.course.id)
    ...

Reply:
good 👁️ 🦌